agent/reconciler/host_reconciler.go

```diff
@@ -156,6 +170,33 @@ func (r *HostReconciler) reconcileNormal(ctx context.Context, byoHost *infrastru
 	return ctrl.Result{}, nil
 }
 
+func (r *HostReconciler) executeInstallerController(ctx context.Context, byoHost *infrastructurev1beta1.ByoHost) error {
+	logger := ctrl.LoggerFrom(ctx)
+	secret := &corev1.Secret{}
+	err := r.Client.Get(ctx, types.NamespacedName{Name: byoHost.Spec.InstallationSecret.Name, Namespace: byoHost.Spec.InstallationSecret.Namespace}, secret)
+	if err != nil {
+		logger.Error(err, "error getting install and uninstall script")
+		r.Recorder.Eventf(byoHost, corev1.EventTypeWarning, "ReadInstallationSecretFailed", "install and uninstall script %s not found", byoHost.Spec.InstallationSecret.Name)
+		return err
+	}
+	installScript := string(secret.Data["install"])
+	uninstallScript := string(secret.Data["uninstall"])
+
+	byoHost.Spec.UninstallationScript = &uninstallScript
+	installScript, err = r.parseScript(ctx, installScript)
+	if err != nil {
+		return err
+	}
+	logger.Info("executing install script")
+	err = r.executeScript(ctx, installScript)
```

Comment:
Can you please reuse the existing utility:
https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/b09a31d78eb0b2e824f3f98dad96f2ead0f638a8/agent/cloudinit/cmd_runner.go#L21-L26

Reply:
@mayur-tolexo I tried with this logic but we have to run command with context. When I reused this utility it was not working.

Reply:
Existing agent installation is not using CommandContext. So, if we execute these commands without context then this shouldn't be breaking anything (+1 for context execution but we can fix that overall in separate PR if required.)